spec/client_nkeys_connect_spec.rb

```diff
@@ -150,5 +195,50 @@
       nats.close
       expect(msgs.count).to eql(1)
     end
+
+    it 'should support user supplied nkey callbacks' do
+      mon = Monitor.new
+      done = mon.new_cond
+
+      errors = []
+      msgs = []
+      nats = NATS::IO::Client.new
+      nats.on_error do |e|
+        errors << e
+      end
+
+      user_nkey_called = false
+      user_nkey_cb = proc {
+        user_nkey_called = true
+        nats.send(:nkey_cb_for_nkey_file, "./spec/configs/nkeys/foo-user.nk").call()
```

Comment:
clearly poking at private functions not a good idea, but seems fine for tests to avoid big code duplication :)